watchtower/wtclient/client.go

```diff
@@ -884,6 +984,182 @@ func (c *TowerClient) initActiveQueue(s *wtdb.ClientSession) *sessionQueue {
 	return sq
 }
 
+// AddTower adds a new watchtower reachable at the given address and considers
+// it for new sessions. If the watchtower already exists, then any new addresses
+// included will be considered when dialing it for session negotiations and
+// backups.
+func (c *TowerClient) AddTower(addr *lnwire.NetAddress) error {
+	tower, err := c.cfg.DB.CreateTower(addr)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case c.newTowers <- &newTowerMsg{tower}:
+	case <-c.pipeline.quit:
+		return ErrClientExiting
+	case <-c.pipeline.forceQuit:
+		return ErrClientExiting
+	}
+
+	return nil
+}
+
+// handleNewTower updates the client's in-memory state to take into account the
+// tower being added.
+//
+// NOTE: This assumes that a call to DB.CreateTower has been made first.
+func (c *TowerClient) handleNewTower(msg *newTowerMsg) {
+	// Update the candidate iterator with the new tower. This might not
+	// actually be a new tower, but it might include a new address at which
+	// it can be reached.
+	c.candidateTowers.AddCandidate(msg.Tower)
+
+	// Include all of its corresponding sessions to our set of candidates.
+	sessions, err := c.cfg.DB.ListClientSessions(&msg.ID)
+	if err != nil {
+		log.Errorf("Unable to determine sessions for tower %x: %v",
+			msg.IdentityKey.SerializeCompressed(), err)
+	}
+	for id, session := range sessions {
+		c.candidateSessions[id] = session
+	}
+}
+
+// RemoveTower removes a watchtower from being considered for future session
+// negotiations and from being used for any subsequent backups until it's added
+// again. If an address is provided, then this call only serves as a way of
+// removing the address from the watchtower instead.
+func (c *TowerClient) RemoveTower(pubKey *btcec.PublicKey, addr net.Addr) error {
+	// We'll load the tower before potentially removing it in order to
+	// retrieve its ID within the database.
+	tower, err := c.cfg.DB.LoadTower(pubKey)
```

Comment:
i think we want to do these calls in the main event loop, so that they can't be interleaved with concurrent operations

Reply:
Done.